proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala

```diff
@@ -620,7 +629,11 @@ object HttpApi {
     }
   }
 
-  final def serve(services: List[(ServiceDescriptor, PartialFunction[HttpRequest, Source[ProtobufAny, NotUsed]])])(
+  final def serve(
+      services: List[
+        (ServiceDescriptor, PartialFunction[HttpRequest, Future[(List[HttpHeader], Source[ProtobufAny, NotUsed])]])
+      ]
```

Comment:
Yeah, I think actually what would be better is a big refactor of this, not just with type aliases, but traits, so that things can be documented nicely.